pkgs/top-level/all-packages.nix

```diff
@@ -2141,6 +2143,8 @@ in
     gcc = gcc49; # doesn't build with gcc5
   };
 
+  jp = (callPackage ../development/tools/jp { }).bin // { outputs = [ "bin" ]; };
```

Comment:
We had a discussion about it here https://github.com/NixOS/nixpkgs/pull/16129#discussion_r66704198

> Running a nix-build . -A jp.all or jp.bin got what I wanted but thats not the normal experience > I expect with installing packages.

You don' normally install packages by `nix-build` - it's meant to be used as development tool. 
If you install a package with `nix-env -i` you don't need to specify package output.